app/src/main/java/com/umbrella/umbrella/LoginActivity.java

```diff
@@ -1,11 +1,43 @@
 package com.umbrella.umbrella;
 
+import android.content.Intent;
 import android.support.v7.app.AppCompatActivity;
 import android.os.Bundle;
+import android.view.View;
+import android.widget.EditText;
+import android.widget.TextView;
 
 /**
  * Created by justin on 16/02/18.
+ * Contains the user interface for logging in
  */
 
-public class LoginActivity {
+public class LoginActivity extends AppCompatActivity {
+    private LoginManager loginManager;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_login);
+        loginManager = new LoginManager();
+    }
+
+    public void buttonPress(View view) {
+        EditText user = findViewById(R.id.userInput);
+        EditText pass = findViewById(R.id.passInput);
+        String username = user.getText().toString(),
```

Comment:
Ideally we would create a separate `LoginViewModel` / `LoginPresenter` that gives us access to these data, and does the UI logic. The presenter would take in the data, and expose the interface needed by the activity. It would return the view model, and the activity would just copy the data into the relevant slots (see how we do it in `RegistrationActivity`, though + a Presenter vs mocked data.

That makes it easier to do UI tests, by removing most of the logic from the activity (you can just assert on the data in the view model).